docs/source/export.rst

```diff
@@ -1,10 +1,445 @@
+
+                # code: out2 = self.branch2(x2)
+                permute_1: f32[128, 64] = torch.ops.aten.permute.default(arg2_1, [1, 0]);
+                addmm_1: f32[s1, 64] = torch.ops.aten.addmm.default(arg3_1, arg5_1, permute_1);
+                relu_1: f32[s1, 64] = torch.ops.aten.relu.default(addmm_1);
+                return (relu, relu_1)
+
+    Graph signature: ExportGraphSignature(parameters=['L__self___branch1_0.weight', 'L__self___branch1_0.bias', 'L__self___branch2_0.weight', 'L__self___branch2_0.bias'], buffers=[], user_inputs=['arg4_1', 'arg5_1'], user_outputs=['relu', 'relu_1'], inputs_to_parameters={'arg0_1': 'L__self___branch1_0.weight', 'arg1_1': 'L__self___branch1_0.bias', 'arg2_1': 'L__self___branch2_0.weight', 'arg3_1': 'L__self___branch2_0.bias'}, inputs_to_buffers={}, buffers_to_mutate={}, backward_signature=None, assertion_dep_token=None)
+    Range constraints: {s0: RangeConstraint(min_val=2, max_val=9223372036854775806), s1: RangeConstraint(min_val=2, max_val=9223372036854775806)}
+    Equality constraints: [(InputDim(input_name='arg4_1', dim=0), InputDim(input_name='arg5_1', dim=0))]
+    """
+
+Some additional things to note:
+
+* Through the :func:`torch.export.dynamic_dim` API, we specified the first
+  dimension of each input to be dynamic. Looking at the inputs ``arg4_1`` and
+  ``arg5_1``, they have a symbolic shape of (s0, 64) and (s1, 128), instead of
+  the (32, 64) and (32, 128) shaped tensors that we passed in as example inputs.
+  ``s0`` and ``s1`` are symbols representing that this dimension can be a range
+  of values.
+
+* ``exported_program.range_constraints`` describes the ranges of each symbol
+  appearing in the graph. In this case, we see that ``s0`` and ``s1`` have
+  the range [2, inf]. For technical reasons that are difficult to explain here,
+  they are assumed to be not 0 or 1. This is not a bug, and does not necessarily
+  mean that the exported program will not work for dimensions 0 or 1. See
+  `The 0/1 Specialization Problem <https://docs.google.com/document/d/16VPOa3d-Liikf48teAOmxLc92rgvJdfosIy-yoT38Io/edit?fbclid=IwAR3HNwmmexcitV0pbZm_x1a4ykdXZ9th_eJWK-3hBtVgKnrkmemz6Pm5jRQ#heading=h.ez923tomjvyk>`_
+  for an in-depth discussion of this topic.
+
+* ``exported_program.equality_constraints`` describes which dimensions are
+  required to be equal. Since we specified in the constraints that the first
+  dimension of each argument is equivalent,
+  (``dynamic_dim(example_args[0], 0) == dynamic_dim(example_args[1], 0)``),
+  we see in the equality constraints the tuple specifying that ``arg4_1``
+  dimension 0 and ``arg5_1`` dimension 0 are equal.
+
+
+Serialization
+^^^^^^^^^^^^^
+
+To save the ``ExportedProgram``, users can use the :func:`torch.export.save` and
+:func:`torch.export.load` APIs. It will serialize the ``ExportedProgram`` to a
+Python dataclass, and then save it as a JSON format. The ``state_dict`` will be
+saved to a file separate from the serialized ``ExportedProgram``.
```

Comment:
Maybe we shouldn't expose the internal details of the file format in case (and quite likely) we change it in the future.